contxt/services/nionic.py

```diff
@@ -31,23 +34,73 @@ def __init__(self, auth: Auth, org_slug: str, env: str = "production", **kwargs)
 
     def get_facilities(self) -> List[Facility]:
-        op = Operation(Query)
-        op.facilities().nodes().__fields__(*list(Facility._ContainerTypeMeta__fields.keys()))
+        op = nionic_operations.query.facilities
         return (op + self.run(op)).facilities.nodes
 
     def create_facility(self, data: dict) -> Facility:
-        op = Operation(Mutation)
-        op.create_facility(input={"facility": data}).__fields__("facility")
-        return self.run(op)["data"]["createFacility"]["facility"]
+        op = nionic_operations.mutation.create_facility
+        return self.run(op, data)["data"]["createFacility"]["facility"]
 
     def get_metric_labels(self) -> List[MetricLabel]:
-        op = Operation(Query)
-        op.metric_labels().__fields__(*list(MetricLabel._ContainerTypeMeta__fields.keys()))
+        op = nionic_operations.query.metric_labels
         return self.run(op).get("data").get("metricLabels")
 
-    def get_metric_data(self, label: str, source_id: str) -> List[MetricData]:
-        op = Operation(Query)
-        fields = list(MetricData._ContainerTypeMeta__fields.keys())
-        fields.remove("id")
-        op.metric_data(source_id=source_id, label=label).nodes().__fields__(*fields)
-        return (op + self.run(op)).metric_data.nodes
+    def get_metric_data(self, label: str, facility_id: int, start: str) -> List[Any]:
+        query = """
+        query metricData($facilityId: Int!, $label: String!, $from: String!) {
+            facility(id: $facilityId) {
+                metricData(label: $label, from: $from, orderBy: TIME_ASC, window: "1 day") {
+                    nodes {
+                        time
+                        data
+                    }
+                }
+            }
+        }
+        """
+        resp = self.query(query, {"facilityId": int(facility_id), "label": label, "from": start})
+        return resp["facility"]["metricData"]["nodes"]
+
+    def get_main_services(
+        self, facility_id: int, resource_type: Optional[ResourceType] = None
+    ) -> List[MainService]:
+        op = nionic_operations.query.main_services
+        filterValues = {"facilityId": int(facility_id)}
+        return [
+            x
+            for x in (op + self.run(op, variables=filterValues)).main_services.nodes
+            if resource_type is None or x.type.lower() == resource_type.value
+        ]
+
+    def get_data_point_data(
+        self, name: str, data_source_name: str, start: str, end: str, window: Window, per_page: int
+    ) -> Iterable[DataPoint]:
+        return PagedGQLTimeSeries(
+            api=self,
+            query="""
+            query dataPointData($dataSourceName: String!, $name: String!,
+            $from: String!, $to: String!, $window: String!, $limit: Int!, $after: Cursor) {
+                dataPoint(dataSourceName: $dataSourceName, name: $name) {
+                    data(from: $from, to: $to, orderBy: TIME_DESC, window: $window
```

Comment:
We may need to add sampling window, use the smaller of 1 day or window?

Reply:
I found that for datapoints we still have a default of 1day though you can ask for data in a smaller window, which results in a single entry in data.